packages/discovery-provider/src/queries/download_csv.py

```diff
@@ -227,56 +250,56 @@ def format_sale_for_download(
     }
 
     if is_for_json_response:
-        # Replace spaces with underscores and add encrypted_email
+        # Replace spaces with underscores and add encrypted data
         fields_with_underscores = {
             key.replace(" ", "_"): value for key, value in base_fields.items()
         }
         fields_with_underscores["encrypted_email"] = (
             result.encrypted_email if hasattr(result, "encrypted_email") else None
         )
+        fields_with_underscores["encrypted_key"] = (
+            result.encrypted_key if hasattr(result, "encrypted_key") else None
+        )
+        fields_with_underscores["buyer_user_id"] = (
+            result.buyer_user_id if hasattr(result, "buyer_user_id") else None
+        )
         return fields_with_underscores
 
     return base_fields
 
 
-# Returns USDC sales for a given artist in a CSV format
 def download_sales(args: DownloadSalesArgs, return_json: bool = False):
+    """Returns USDC sales for a given artist in CSV or JSON format."""
     seller_user_id = args["seller_user_id"]
+    grantee_user_id = args.get("grantee_user_id")
 
     # Get sales for artist
-    results = get_purchases_or_sales(seller_user_id, is_purchases=False)
+    results = get_purchases_or_sales(
+        seller_user_id, is_purchases=False, grantee_user_id=grantee_user_id
+    )
 
-    # Get artist handle and encryption key
+    # Get artist handle
     db = get_db_read_replica()
     with db.scoped_session() as session:
-        seller_data = (
-            session.query(User.handle, EmailEncryptionKey.encrypted_key)
+        seller = (
+            session.query(User.handle)
             .filter(User.user_id == seller_user_id)
             .filter(User.is_current == True)
-            .outerjoin(
-                EmailEncryptionKey, EmailEncryptionKey.primary_user_id == User.user_id
-            )
             .first()
         )
+        seller_handle = seller.handle if seller else None
```

Comment:
if there's no seller here, we'll generate an empty response? maybe worth just throwing in that case? can be a todo